organization-bootstrap/README.md

```diff
@@ -0,0 +1,46 @@
+
+There are several advantages in using an initial stage like the ones provided here:
+
+- automate and parameterize creation of the organizational layout, documenting it through code
+- use a single declarative tool to create and manage both prerequisites and the actual infrastructure, eliminating the need for manual commands, scripts or external tools
+- enforce separation of duties at the environment (or tenant in multi-tenant architectures) level, by automating creation of per-environment Terraform service accounts, their IAM roles, and GCS buckets to hold state
+- decouple and document the use of organization-level permissions from the day to day management of the actual infrastructure, by assigning a minimum but sufficient set of high level IAM roles to Terraform service accounts in an initial stage
+- provide a sane place for the creation and management of shared resources that are not tied to a specific environment
+
+## Operational considerations
+
+This specific type of preliminary automation stage is usually fairly static, only changing when a new environment or shared resource is added or modified, and lends itself well to being applied manually by organization administrators. One secondary advantage of running this initial stage manually, is eliminating the need to create and manage automation credentials that embed sensitive permissions scoped at the organization or root folder level.
+
+### IAM roles
+
+This type of automation stage needs very specific IAM roles on the root node (organization or folder), and additional roles at the organization level if the generated service accounts for automation need to be able to create and manage Shared VPC. The needed roles are:
+
+- on the root node Project Creator, Folder Administrator, Logging Administrator
+- on the billing account or organization Billing Account Administrator
+- on the organization Organization Administrator, if Shared VPC needs to be managed by the automation service accounts
+
+### State
+
+This type of stage creates the prerequisites for Terraform automation including the GCS bucket used for its own remote state, so some care needs to be used when running it for the first time, when its GCS bucket has not yet been created. This is the sequence of steps to be followed to enable remote state management:
+
+- when running `apply` for the first time, the `backend.tf` file needs to be commented so local state is used
+- after the first `apply` has completed successfully, the comments in the `backend.tf` file need to be removed, and the GCS bucket name from the `bootstrap_tf_gcs_bucket` output added as a value for the `bucket` attribute
+  - in the `backedn.tf` file directly, or better
+  - in a separate `backend-config.hcl` file that will then be passed in the `-backend-config` option to every `init` call
```

Comment:
I'd strongly discourage this approach, it's all too easy to forget to include this one time and thereby corrupt everything. Much better to be explicit and include it in backend.tf. Curious on why we'd recommend this?

Reply:
Explicit meaning you'd include the bucket name? And how do you then create the bucket?

We've always used this approach and never had an issue in the ~10 actual installs we've had so far (and the countless test installs we do for testing, prototyping, etc). I like it because you pay a minimal price once, and then are set with everything documented in code.